assignments/array-methods.js

```diff
@@ -55,29 +55,58 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
+
 let fullName = [];
+runners.forEach(function(elem) {
+    fullName.push(`${elem.first_name} ${elem.last_name}`)
+});
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
+
+let allCaps = runners.map(function (elem) {
+return elem.first_name.toUpperCase();
+})
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+
+let largeShirts = runners.filter(function(elem) {
+return elem.shirt_size === "L";
+})
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+
+let ticketPriceTotal = runners.reduce(function(total, elem) {
+return total += elem.donation;
+}, 0)
 console.log(ticketPriceTotal);
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to create and then solve 3 unique problems using one or many of the array methods listed above.
 
-// Problem 1
+// Problem 1: The director is awarding medals to those who donated over 200, return info about people who donated over 200.
+
+let overAchievers = runners.filter(function (elem){
+return elem.donation >= 200
+})
+console.log(overAchievers)
+
+// Problem 2: The director is giving prices to the 35th person to sign up. Return an array containing the information of the 35th id-holder in the given array.
+
+let luckyRunner = runners.filter(function (elem) {
+return elem.id === 35
+})
+console.log(luckyRunner)
 
-// Problem 2
+// Problem 3: The director really wants to see the amount of donations from each company. Return an array containing companies with their donation amounts.
 
-// Problem 3
\ No newline at end of file
+let greedArray = [];
+runners.forEach(function (elem) {
+greedArray.push(`${elem.company_name} donated $${elem.donation}`)
+})
```

Comment:
When we use `push` in a `forEach` and nothing else, that is usually a signal that we want to use a `map` instead, as that is its purpose. We could have also used a `reduce` to collect companies and donations in a object, in case multiple runners worked for the same company.